check.go

```diff
@@ -268,6 +284,27 @@ func (c *CheckRunner) UpdateCheck(checkID types.CheckID, status, output string)
 		return
 	}
 
+	// Do nothing if update is idempotent
+	if check.Status == status && check.Output == output {
+		check.failuresCounter = 0
+		check.successCounter = 0
+		return
+	}
+
+	if status == api.HealthCritical {
+		check.failuresCounter++
+		if check.failuresCounter < c.FailuresBeforeCritical {
+			check.failuresCounter = 0
+			return
+		}
+	} else {
+		check.successCounter++
+		if check.successCounter < c.SuccessBeforePassing {
+			check.successCounter = 0
+			return
+		}
+	}
```

Comment:
As I understand this, when `c.FailuresBeforeCritical ` and `c.SuccessBeforePassing` are values greater than 1, the counters will perpetually increment and reset to 0, so that the threshold will never be crossed and the status will never toggle over.

Testing with `failures_before_critical` `success_before_passing` equal to 2 confirms this for me - the health check doesn't switch its status. Let me know if I'm missing something or can elaborate further

Reply:
The idea is that the health check only switches state when X `consecutive` failures or successes have occurred. So if you have FailuresBeforeCritical == 3, current status is passing, and you get:

fail, fail, pass, fail, fail, pass, fail, fail, pass. 

The the status will never switch, because you never had 3 consecutive failures. This was the original intent. However, I have changed the reset logic to instead decrement the current counter. Now, it would trigger on the forth fail. I feel that this behavior is more akin to what one would want in production.

Reply:
I also fixed a bug that I had introduced while trying to simplify the check logic. That may have induced you in error.

Reply:
I added a test, and the behavior seems to be as expected. Do you want me to test a specific situation?

Reply:
Thank you for adding this test! No specific situation - the test was very clarifying. I agree with you that the decrement and behavior is as expected.